### PR TITLE
fix(utilities): refactor utilities, add tests, typo fixes

### PR DESCRIPTION
**Pull request by maghirardelli**
Issue #, if available:

Description of changes:
refactor utilities for pagination and dynamoDBService and services that consume them
add tests for DynamoDB
fix small typos

Checklist:

* [x] Have you successfully deployed to an AWS account with your changes?
* [x] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by aws-tyler — CHANGES_REQUESTED** [review]
Your code looks great. A potential vuln was found though, so marking as `Request Changes` until we decide whether to fix it or leave it as is for now to keep us unblocked. If we leave it for now we need to make a ticket to go back and fix ASAP.

**Review by bhudnell — APPROVED** [review]
Looks like the CodeQL alert is a false positive to me. I'm guessing that it saw `params` and assumed it was from a request object liek `req.params.variableName`

**Review by FernandoAranda — APPROVED** [review]

File: workbench-core/base/src/constants/queryParams.ts
Comment (FernandoAranda) [inline workbench-core/base/src/constants/queryParams.ts:6]: this will affect the changes for filter and sort, I will set a reminder for myself to check this after merging our common code

**Comment by nguyen102** [comment]
> Your code looks great. A potential vuln was found though, so marking as `Request Changes` until we decide whether to fix it or leave it as is for now to keep us unblocked. If we leave it for now we need to make a ticket to go back and fix ASAP.

To resolve the security vulnerability, we can check the length of `params.datasetIds` and make sure it's less than N items. DDB query can only return 1 MB of data so if there's too many `datasetIds` we would need to query DDB repeatedly. We can check how much space an additional `datasetId` requires (for each datasetId, we pull the dataset record from DDB [here](https://github.com/aws-solutions/solution-spark-on-aws/pull/592/files#diff-f1abf758c3206eeb07deec5194b081e75de50255015a900efcd2f07a7a038448R384) and set N appropriately. If `datasetIds` is larger than N we can throw a `Boom.badRequest` error.

**Comment by maghirardelli** [comment]
I fixed the CodeQL Warning: https://github.com/aws-solutions/solution-spark-on-aws/security/code-scanning/49. @aws-tyler can you unblock this PR and approve?

**Review by aws-tyler — APPROVED** [review]

**Review by DaveMadsen — APPROVED** [review]

**Pull request merged**
